kroxylicious-kms-provider-aws-kms/src/main/java/io/kroxylicious/kms/provider/aws/kms/AwsV4SigningHttpRequestBuilder.java

```diff
@@ -0,0 +1,364 @@
+/*
+ * Copyright Kroxylicious Authors.
+ *
+ * Licensed under the Apache Software License version 2.0, available at http://www.apache.org/licenses/LICENSE-2.0
+ */
+
+package io.kroxylicious.kms.provider.aws.kms;
+
+import java.net.URI;
+import java.net.http.HttpClient;
+import java.net.http.HttpRequest;
+import java.nio.ByteBuffer;
+import java.nio.charset.StandardCharsets;
+import java.security.InvalidKeyException;
+import java.security.MessageDigest;
+import java.security.NoSuchAlgorithmException;
+import java.time.Duration;
+import java.time.Instant;
+import java.time.ZoneOffset;
+import java.time.format.DateTimeFormatter;
+import java.util.ArrayList;
+import java.util.Comparator;
+import java.util.HashMap;
+import java.util.HexFormat;
+import java.util.Locale;
+import java.util.Map;
+import java.util.Objects;
+import java.util.Optional;
+import java.util.concurrent.Flow;
+import java.util.function.Predicate;
+import java.util.stream.Collectors;
+
+import javax.crypto.Mac;
+import javax.crypto.spec.SecretKeySpec;
+
+import io.kroxylicious.kms.service.KmsException;
+import io.kroxylicious.proxy.tag.VisibleForTesting;
+
+import edu.umd.cs.findbugs.annotations.NonNull;
+
+import static java.net.http.HttpRequest.BodyPublisher;
+import static java.net.http.HttpRequest.Builder;
+
+/**
+ * An implementation of HttpRequestBuilder that signs AWS requests
+ * accordance with <a href="https://docs.aws.amazon.com/IAM/latest/UserGuide/create-signed-request.html">AWS v4</a>.
+ */
+class AwsV4SigningHttpRequestBuilder implements Builder {
+
+    private static final DateTimeFormatter DATE_TIME_FORMATTER = DateTimeFormatter.ofPattern("yyyyMMdd'T'HHmmss'Z'").withZone(ZoneOffset.UTC);
```

Comment:
Isn't there a factory method for this pattern somewhere in `java.time` API already?

Reply:
Unfortunately, no. AWS requires the date/time formatted without the hyphens and colons that DateTimeFormatter.ISO_INSTANT provides.  I don't see a way on DateTimeFormatter to turn off those separators.

https://docs.aws.amazon.com/IAM/latest/UserGuide/create-signed-request.html